basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -40,159 +46,187 @@
         ResourceAddress address = hostTemplate.resolve(context, httpServer, hostName);
         executeAddAction(address);
         return hostName;
     }
 
-    public void removeHTTPServerHost(String httpServer, String ajpListener) {
+    public void removeHTTPServerHost(String httpServer, String ajpListener) throws InterruptedException, TimeoutException, IOException {
         ResourceAddress address = hostTemplate.resolve(context, httpServer, ajpListener);
         executeRemoveAction(address);
     }
 
-    public void removeHTTPServerHostIfExists(String httpServer, String ajpListener) {
+    public void removeHTTPServerHostIfExists(String httpServer, String ajpListener) throws InterruptedException, TimeoutException, IOException {
         ResourceAddress address = hostTemplate.resolve(context, httpServer, ajpListener);
         if (executeReadResourceAction(address)) {
             executeRemoveAction(address);
         }
     }
 
-    public String createHTTPServer() {
+    public String createHTTPServer() throws InterruptedException, IOException, TimeoutException {
         String name = RandomStringUtils.randomAlphanumeric(6);
         ResourceAddress address = httpServerTemplate.resolve(context, name);
-        executeAddAction(address);
+        executeAddAction(address, true);
+        operations.readResource(Address.of("subsystem", "undertow").and("server", name)); //Workaround for situation when server is listed as unavailable dependency
         return name;
     }
 
-    public void removeHTTPServer(String httpServer) {
+    public void removeHTTPServer(String httpServer) throws InterruptedException, TimeoutException, IOException {
         executeRemoveAction(httpServerTemplate.resolve(context, httpServer));
     }
 
-    public String createServletContainer() {
+    public String createServletContainer() throws InterruptedException, TimeoutException, IOException {
         String name = RandomStringUtils.randomAlphanumeric(6);
         ResourceAddress address = servletContainerTemplate.resolve(context, name);
         executeAddAction(address);
         return name;
     }
 
-    public void removeServletContainer(String servletContainer) {
+    public void removeServletContainer(String servletContainer) throws InterruptedException, TimeoutException, IOException {
         executeRemoveAction(servletContainerTemplate.resolve(context, servletContainer));
     }
 
-    private void executeRemoveAction(ResourceAddress address) {
-        dispatcher.execute(new Operation.Builder("remove", address).build());
+    private void executeRemoveAction(ResourceAddress address) throws InterruptedException, IOException, TimeoutException {
+        Dispatcher dispatcher = new Dispatcher();
+        dispatcher.execute(new Operation.Builder("remove", address).build()); //TODO add creaper to all methods, then remove
+        dispatcher.close();
```

Comment:
I prefer to use `try` and close `Dispatcher` in `finally` block but as you mentioned this code has to be refactored to Creaper soon so I'm ok with that.